Sources/SupabaseStorage/StorageBucketApi.swift

```diff
@@ -10,93 +10,167 @@ public class StorageBucketApi: StorageApi {
+
+    let body = Bucket.CreateRequest(id: id, isPublic: isPublic)
+
+    let responseData = try await fetch(
+      url: url,
+      method: .put,
+      json: body,
+      headers: headers
+    )
+
+    let responseObject = try decoder.decode(OneKeyedResponse<String>.self, from: responseData)
+
+    guard responseObject.key == "message", responseObject.value == "Successfully updated" else {
+      throw StorageError(message: "'\(responseObject.key)' = '\(responseObject.value)'")
     }
+  }
 
-    let response = try await fetch(url: url, method: .post, parameters: [:], headers: headers)
-    guard let dict = response as? [String: Any] else {
-      throw StorageError(message: "failed to parse response")
+  /// Removes all objects inside a single bucket.
+  ///
+  /// > RLS policy permissions required:
+  /// > - `buckets` table permissions: `select`
+  /// > - `objects` table permissions: `select` and `delete`
+  ///
+  /// - Parameter id: The unique identifier of the bucket you would like to empty.
+  public func emptyBucket(id: String) async throws {
+    let url = url.appendingPathComponent("/bucket/\(id)/empty")
+
+    let responseData = try await fetch(
+      url: url,
+      method: .post
+    )
+
+    let responseObject = try decoder.decode(OneKeyedResponse<String>.self, from: responseData)
+
+    if responseObject.key != "message" {
+      throw StorageError(message: "failed to parse response - missing 'message'")
+    }
+
+    if responseObject.value != "Successfully emptied" {
+      throw StorageError(message: "response error: \(responseObject.value)")
     }
-    return dict
   }
 
   /// Deletes an existing bucket. A bucket can't be deleted with existing objects inside it.
-  /// You must first `empty()` the bucket.
+  ///
+  /// > You must first empty the bucket using ``emptyBucket(id:)``
+  ///
+  /// > RLS policy permissions required:
+  /// > - `buckets` table permissions: `select` and `delete`
+  /// > - `objects` table permissions: none
+  ///
   /// - Parameters:
   ///   - id: The unique identifier of the bucket you would like to delete.
-  public func deleteBucket(id: String) async throws -> [String: Any] {
-    guard let url = URL(string: "\(url)/bucket/\(id)") else {
-      throw StorageError(message: "badURL")
-    }
+  public func deleteBucket(id: String) async throws {
+    let url = url.appendingPathComponent("/bucket/\(id)")
 
-    let response = try await fetch(url: url, method: .delete, parameters: [:], headers: headers)
-    guard let dict = response as? [String: Any] else {
+    let responseData = try await fetch(
+      url: url,
+      method: .delete
+    )
+
+    let responseObject = try decoder.decode(OneKeyedResponse<String>.self, from: responseData)
+
+    if responseObject.key == "id" {
       throw StorageError(message: "failed to parse response")
     }
-    return dict
+
+    if responseObject.value != "Successfully deleted" {
+      throw StorageError(message: "unknown error")
+    }
```

Comment:
Also, I don't know how exactly these endpoints work, but it seems a code smell to compare strings for deciding something.